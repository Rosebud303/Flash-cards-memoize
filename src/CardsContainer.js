import React, { Component } from 'react';
import Cards from './Cards.js'
import './_CardsContainer.scss'

export default class CardsContainer extends Component {
    constructor(props) {
        super(props);
        this.state = {
            toggle: false,
            deck: [],
            reset: false
        }
    }

    componentDidMount (){
        this.setState({
            deck: this.props.prototypes
        })
    }

    toggleReset = () => {
        this.setState({
            reset: false
        })
    }

    toggleWrong = () => {
        if(this.state.toggle){
            this.setState({
                deck: this.props.prototypes,
                toggle: false
            })
        }else{
            let filteredDeck = this.props.prototypes.filter((card) => {
                return this.props.wrong.includes(card.id)
            })
            this.setState({
                deck: filteredDeck,
                toggle: true
            })
        }
    }
    render(){
        console.log(this.state.deck)
        return (
            <div className="card-container">
                    <h2>PROTOTYPE.memoizeâ„¢</h2>
                    <h3>Wrong Count: {this.props.wrong.length}</h3>
                    <button className='show' onClick={this.toggleWrong}>Show Wrong</button>
                    <button className='show' onClick={this.toggleReset}>Reset</button>
                <ul className='cards'>
                    {
                        this.state.deck.map( proto => {
                            return <Cards key={proto.id} 
                                        proto={proto}
                                        wrong={this.props.wrong}
                                        reset={this.state.reset}
                                    />
                        })
                    }
                </ul>
            </div>
        )
    }
 
}