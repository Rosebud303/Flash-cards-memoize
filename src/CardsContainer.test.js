import React from 'react';
import CardsContainer from './CardsContainer.js';
import { shallow } from 'enzyme';

const mockArray = [{
    "id": 1,
    "name": ".concat",
    "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
    "type": "accessor"
  },
  {
    "id": 2,
    "name": ".copyWithin",
    "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
    "type": "mutator"
  },
  {
    "id": 3,
    "name": ".entires",
    "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
    "type": "iteration"
  }
]

const newMockArray = [1,2,3]

describe('CardsContainer', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
                <CardsContainer prototypes={mockArray}
                    wrong={mockArray}
                />
       )
    });

    it('should create an exact snapshot of component', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    it('should have default state', () => {
        expect(wrapper.state()).toEqual({toggle: false, deck: mockArray, reset: false});
    });

    it('should setState to false when invoked', () =>{
        expect(wrapper.state("reset")).toEqual(false);
        wrapper.instance().toggleReset()
        expect(wrapper.state("reset")).toEqual(false);        
    })
})