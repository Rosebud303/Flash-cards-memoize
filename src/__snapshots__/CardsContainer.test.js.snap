// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should create an exact snapshot of component 1`] = `
<div
  className="card-container"
>
  <h2>
    PROTOTYPE.memoizeâ„¢
  </h2>
  <h3>
    Wrong Count: 
    3
  </h3>
  <button
    className="show"
    onClick={[Function]}
  >
    Show Wrong
  </button>
  <button
    className="show"
    onClick={[Function]}
  >
    Reset
  </button>
  <ul
    className="cards"
  >
    <Cards
      key="1"
      proto={
        Object {
          "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
          "id": 1,
          "name": ".concat",
          "type": "accessor",
        }
      }
      reset={false}
      wrong={
        Array [
          Object {
            "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
            "id": 1,
            "name": ".concat",
            "type": "accessor",
          },
          Object {
            "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
            "id": 2,
            "name": ".copyWithin",
            "type": "mutator",
          },
          Object {
            "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
            "id": 3,
            "name": ".entires",
            "type": "iteration",
          },
        ]
      }
    />
    <Cards
      key="2"
      proto={
        Object {
          "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
          "id": 2,
          "name": ".copyWithin",
          "type": "mutator",
        }
      }
      reset={false}
      wrong={
        Array [
          Object {
            "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
            "id": 1,
            "name": ".concat",
            "type": "accessor",
          },
          Object {
            "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
            "id": 2,
            "name": ".copyWithin",
            "type": "mutator",
          },
          Object {
            "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
            "id": 3,
            "name": ".entires",
            "type": "iteration",
          },
        ]
      }
    />
    <Cards
      key="3"
      proto={
        Object {
          "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
          "id": 3,
          "name": ".entires",
          "type": "iteration",
        }
      }
      reset={false}
      wrong={
        Array [
          Object {
            "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
            "id": 1,
            "name": ".concat",
            "type": "accessor",
          },
          Object {
            "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
            "id": 2,
            "name": ".copyWithin",
            "type": "mutator",
          },
          Object {
            "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
            "id": 3,
            "name": ".entires",
            "type": "iteration",
          },
        ]
      }
    />
  </ul>
</div>
`;
